src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -117,6 +117,21 @@ public function normalize($object, $format = null, array $context = [])
                 $attributeValue = $callback($attributeValue, $object, $attribute, $format, $context);
             }
 
+            if (isset($attributesMetadata[$attribute]) && ($embedProperties = \array_keys($attributesMetadata[$attribute]->getEmbedProperties()))) {
```

Comment:
This looks wrong...
```suggestion
            if (isset($attributesMetadata[$attribute])) {
                foreach ($attributesMetadata[$attribute]->getEmbedProperties()) as $embedProperty => $value) {
                    // Retrieve all embed properties
                    $data = $this->updateData(
                        $data,
                        $embedProperty,
                        $this->getAttributeValue($attributeValue, $embedProperty, $format, $context),
                        \get_class($attributeValue),
                        $format,
                        $context
                    );

                    continue 2;
                }
            }
```

Reply:
No, with the `continue 2` in the loop, we cannot retrieve all embed properties.